src/platform/plugins/private/image_embeddable/public/image_embeddable/get_image_embeddable_factory.tsx

```diff
@@ -30,72 +29,80 @@ export const getImageEmbeddableFactory = ({
+      const dynamicActionsManager = embeddableEnhanced?.initializeEmbeddableDynamicActions(
         uuid,
         () => titleManager.api.title$.getValue(),
-        initialState
+        initialState.rawState
       );
       // if it is provided, start the dynamic actions manager
-      const maybeStopDynamicActions = dynamicActionsApi?.startDynamicActions();
+      const maybeStopDynamicActions = dynamicActionsManager?.startDynamicActions();
 
       const filesClient = filesService.filesClientFactory.asUnscoped<FileImageMetadata>();
-      const imageConfig$ = new BehaviorSubject<ImageConfig>(initialState.imageConfig);
+      const imageConfig$ = new BehaviorSubject<ImageConfig>(initialState.rawState.imageConfig);
       const dataLoading$ = new BehaviorSubject<boolean | undefined>(true);
 
-      const embeddable = buildApi(
-        {
-          ...titleManager.api,
-          ...(dynamicActionsApi?.dynamicActionsApi ?? {}),
-          dataLoading$,
-          supportedTriggers: () => [IMAGE_CLICK_TRIGGER],
-          onEdit: async () => {
-            try {
-              const newImageConfig = await openImageEditor({
-                parentApi: embeddable.parentApi as PresentationContainer,
-                initialImageConfig: imageConfig$.getValue(),
-              });
-              imageConfig$.next(newImageConfig);
-            } catch {
-              // swallow the rejection, since this just means the user closed without saving
-            }
-          },
-          isEditingEnabled: () => true,
-          getTypeDisplayName: () =>
-            i18n.translate('imageEmbeddable.imageEmbeddableFactory.displayName.edit', {
-              defaultMessage: 'image',
-            }),
-          serializeState: () => {
-            return {
-              rawState: {
-                ...titleManager.serialize(),
-                ...(dynamicActionsApi?.serializeDynamicActions() ?? {}),
-                imageConfig: imageConfig$.getValue(),
-              },
-            };
+      function serializeState() {
+        return {
+          rawState: {
+            ...titleManager.getLatestState(),
+            ...(dynamicActionsManager?.getLatestState() ?? {}),
+            imageConfig: imageConfig$.getValue(),
           },
+        };
+      }
+
+      const unsavedChangesApi = initializeUnsavedChanges<ImageEmbeddableSerializedState>({
+        uuid,
+        parentApi,
+        serializeState,
+        anyStateChange$: merge(
+          titleManager.anyStateChange$,
+          imageConfig$.pipe(map(() => undefined))
+        ),
+        getComparators: () => {
+          return {
+            ...(dynamicActionsManager?.comparators ?? { enhancements: 'skip' }),
+            ...titleComparators,
+            imageConfig: 'deepEquality',
+          };
+        },
+        onReset: (lastSaved) => {
+          titleManager.reinitializeState(lastSaved?.rawState);
```

Comment:
Looks like resetting the dashboard is not clearing unsaved state for an image embeddable. I tested this in `main` and it is working.

https://github.com/user-attachments/assets/4d10e596-8ffd-489b-a9de-285ff5f3755f

Reply:
Thanks for testing the image embeddable and finding this regression. This is resolved with 83762e9da7af91175413f0b46dd2f18815cecf4a